hibernate-core/src/test/java/org/hibernate/test/cfg/NumberPropertyConversionTest.java

```diff
@@ -0,0 +1,24 @@
+package org.hibernate.test.cfg;
+
+import java.util.Collections;
+
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Environment;
+import org.hibernate.engine.jdbc.spi.JdbcServices;
+import org.hibernate.service.ServiceRegistry;
+
+import org.hibernate.testing.boot.ServiceRegistryTestingImpl;
+import org.hibernate.testing.junit4.BaseUnitTestCase;
+import org.junit.Test;
+
+public class NumberPropertyConversionTest extends BaseUnitTestCase {
+
+	@Test
+	public void testIntegerToLongConversion() {
+		ServiceRegistry sr = ServiceRegistryTestingImpl.forUnitTesting(
+				Collections.singletonMap( Environment.LOG_SLOW_QUERY, 25 )
```

Comment:
Technically, using `Environment` is okay for it inherits `AvailableSettings`, but using `AvailableSettings.LOG_SLOW_QUERY` seems still better.
Also from testing perspective, you might go further than ensuring launching is not broken by adding the following assertions:

1. assert the config's original intended type is 'long', not 'int';
2. assert that the value after conversion is as intended (i.e. we received `25L`).

You also need to add `@TestForIssue( jiraKey = "HHH-13893" )` for this method.

Reply:
This test is simplified "replication" of problem that I've met in one real situation. I think that it can be completely removed since suggested changes can be simply tested by directly testing ConfigurationHelper. I've created test case in PropertiesHelperTest that will, hopefully, better handle this issue.